src/boundary_condition.cpp

```diff
@@ -16,7 +17,18 @@ namespace openmc {
 
 void VacuumBC::handle_particle(Particle& p, const Surface& surf) const
 {
-  p.cross_vacuum_bc(surf);
+  // Random ray and Monte Carlo need different treatments at vacuum BCs
+  if (p.type() == ParticleType::random_ray) {
+    // Reflect ray off of the surface
+    ReflectiveBC().handle_particle(p, surf);
+
+    // Set ray's angular flux spectrum to vacuum conditions (zero)
+    RandomRay* r = static_cast<RandomRay*>(&p);
```

Comment:
Having boundary conditions as classes just really does not make a ton of sense... I wish it had just been an enum code, then we would just have a BC handling method on each particle type that takes as an input the code and modifies the internal state. Oh well, we're already here!